hack/generator/pkg/astmodel/package_reference.go

```diff
@@ -5,31 +5,65 @@
 
 package astmodel
 
-import "path/filepath"
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	localPathPrefix = "github.com/Azure/k8s-infra/hack/generator/apis/"
+)
 
 // PackageReference indicates which package
 // a struct belongs to.
 type PackageReference struct {
-	groupName   string
-	packageName string
+	packagePath string
 }
 
-// PackagePath is the path to the package reference
-func (pr *PackageReference) PackagePath() string {
-	return filepath.Join(pr.GroupName(), pr.PackageName())
+// NewLocalPackageReference creates a new local package reference
+func NewLocalPackageReference(groupName string, packageName string) PackageReference {
+	return PackageReference{
+		packagePath: filepath.Join(localPathPrefix, groupName, packageName),
+	}
+}
+
+func (pr *PackageReference) isLocalPackage() bool {
+	return strings.HasPrefix(pr.packagePath, localPathPrefix)
 }
 
-// GroupName is the group name of the package reference
-func (pr *PackageReference) GroupName() string {
-	return pr.groupName
+func (pr *PackageReference) stripLocalPackagePrefix() (string, error) {
+	if !pr.isLocalPackage() {
+		return "", fmt.Errorf("cannot strip local package prefix from non-local package %v", pr.packagePath)
```

Comment:
Capitalize `"Cannot strip ...` and quotes `... package '%v'"` #nitpick

Reply:
According to my goland, error strings aren't supposed to begin with an uppercase character (it green-squiggles me if they aren't).

I see here: https://github.com/golang/go/wiki/CodeReviewComments#error-strings -- although it doesn't say why still...

Maybe @devigned knows.

Reply:
I'm not getting those squiggles.

_Sigh_. If lowercase is idiomatic Go, then we should follow that ... but it will take a while before it looks anything other than amateurish to me.

Reply:
Yeah I don't love it either but most of our error strings already seem to be following that